tests/monotouch-test/mono/WeakReferenceTest.cs

```diff
@@ -0,0 +1,110 @@
+﻿//
+// Unit tests for WeakAttribute
+//
+// Authors:
+//	Alex Soto <alexsoto@microsoft.com>
+//
+//
+// Copyright 2018 Xamarin Inc. All rights reserved.
+//
+
+using System;
+
+#if XAMCORE_2_0
+using Foundation;
+using SpriteKit;
+#else
+using MonoTouch;
+using MonoTouch.Foundation;
+using MonoTouch.SpriteKit;
+#endif
+
+using NUnit.Framework;
+
+namespace MonoTouchFixtures {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class WeakReferenceTest {
+		const int totalTestObjects = 100;
+
+		[SetUp]
+		public void Setup ()
+		{
+#if __WATCHOS__
+			// watchOS 3.0+
+			//TestRuntime.CheckWatchOSSystemVersion (3, 0);
+			Assert.Ignore ("WeakAttribute is not working on watchOS yet");
+#else
+			// iOS 7.0+ macOS 10.9+ tvOS 9.0+
+			TestRuntime.AssertXcodeVersion (5, 0);
+#endif
+		}
+
+		[Test]
+		public void NoRetainCyclesExpectedTest ()
+		{
+			for (int i = 0; i < totalTestObjects; i++) {
+				var parent = new MyParentView (useWeak: true);
+				parent.TouchButton ();
+			}
+
+			GC.Collect ();
+			GC.WaitForPendingFinalizers ();
+			Assert.That (MyParentView.count, Is.LessThan (totalTestObjects), "No retain cycles expected");
+		}
+
+		[Test]
+		public void RetainCyclesExpectedTest ()
+		{
+			for (int i = 0; i < totalTestObjects; i++) {
```

Comment:
You're not setting `MyParentView.count` to zero here, which means that its result depends on the result of the other test.

Reply:
it would explain the 
```
[FAIL] WeakReferenceTest.RetainCyclesExpectedTest :   Retain cycles expected
  Expected: 100
  But was:  200
```
on 32 bits :)